Formula/koka.rb

```diff
@@ -0,0 +1,26 @@
+class Koka < Formula
+  desc "Compiler for the Koka language"
+  homepage "http://koka-lang.org"
+  url "https://github.com/koka-lang/koka/releases/download/v2.3.8/koka-v2.3.9-source.tar.gz"
+  sha256 "ae3e8127b46e51cc4df3525c5f913ecfbaeeb29f0a268e4a3d2e0ac809bc987f"
+  license "Apache-2.0"
+  head "https://github.com/koka-lang/koka.git", branch: "master"
+
+  depends_on "cabal-install" => :build
+  depends_on "ghc" => :build
+  depends_on "vcpkg"
+
+  def install
+    system "cabal", "new-update"
```

Comment:
Does this pull in dependencies?

Reply:
We kinda do the same thing elsewhere with `cabal v2-update`. Though now I'm wondering what the difference is between `v2-update` and `new-update`.

Reply:
[I think they're the same](https://stackoverflow.com/questions/66535628/what-is-the-difference-between-install-v1-install-v2-install-and-new-install). TLDR: For modern versions of `cabal`, `v2-*` is the same as `*` or `new-*`. `v1-*` refers explicitly to the older deprecated commands.

Reply:
We should probably use `v2-*` instead of `new-*` to be consistent with other formulae then. We probably also want to use `std_cabal_v2_args`.

Reply:
Thanks for the detailed feedback. The `new-` and `v2-` prefixes are equivalent, and since Cabal version 3 also equal to the plain `update`, `build`, and `run`. I updated to use the plain commands.

I am not using `std_cabal_v2_args` since those are meant for `cabal install` and not for `cabal build`. Koka is not using `cabal install` as it has its own bundler (since it also needs to include the standard libraries etc). 

I think it looks quite good now. Thanks.

Reply:
Thanks for the clarification. What does Koka bundle? Is that why you need `vcpkg`? More generally, what is the `vcpkg` dependency used for?

Reply:
First the Koka compiler is built (using `cabal build`) and then the compiler itself executes the `bundle.kk` script -- this script uses the newly build compiler to compile all standard libraries (in three variants (debug, release, etc)), then tar's everything together into a single tar archive, and 
finally installs the results (to <prefix>/bin <prefix>/lib and <prefix>/share).  

The `vcpkg` dependency is used when building the standard libraries for Koka; some of these use external C libraries and header files, like the `std/text/regex` module links with `pcre2`. When compiling the module Koka invokes `vcpkg` to install/resolve the required `pcre2` library+headers.

Reply:
When is the standard library built? Does that happen when this formula is built, or do users build it for themselves later on?

Reply:
The standard libraries should be built when the formula is built. They are built in 3 variants and it takes quite a while so they are usually distributed with the compiler in binary form (so the compiler is snappy when loading the standard libraries).

Reply:
This is probably now broken on the tip of the `master` branch then, since the `vcpkg` formula now needs configuration to work properly.

In any case, we should be using Homebrew `pcre2` here instead of what's provided by `vcpkg`. Using `vcpkg`-provided `pcre2` sounds like it relies on state that won't survive repackaging. (i.e. `brew install koka` will result in a broken standard library unless you compiled it from source yourself).

Reply:
Actually repackaging will will work fine as koka always links statically and includes the needed .o or .lib files so it will not break if reinstalled in binary form (and the end user needs to install no other libraries).

I thought about including all needed C library dependencies in the brew formula itself ... but that may be troublesome:

1. Currently the Koka modules themselves declare packages they rely on; I would need to ensure manually this matches what I put in the formula.
2. Currently, the Koka compiler knows about vcpkg and Conan -- if we install libraries through Brew we need to probably add some more support for that in the compiler.
3. Koka modules can declare the versions of C libraries they depend on (or what versions should not be used): this is supported by vcpkg and Conan, but not by Brew :-(
4. There are many more C libraries available through vcpkg and Conan than through Brew which seems to more used for end-user applications? 

I can potentially do 1 and 2 (but not ideal and would take some work), but 2 & 3 could prove problematic especially if the particular library is not supported by Brew... (however, it seems I only need pcre2, libuv, and libz so for Koka it may work out.) 

This is a general problem with tools like Brew, or apt, or pkg, etc. which seem geared to end-user applications while vcpkg and Conan are more oriented to C library installation. What is the end-goal of Brew here -- support all C libraries?

Reply:
Ah apologies for the long reply. I guess tldr:

1. Koka has a build dependency on vcpkg (or Conan) to get the exact right versions of some C libraries needed for the standard libraries which copy archives and link statically
2. There is also a install dependency on vcpckg (or Conan) for the user since a user may write a Koka module that needs itself to be linked to some other C libraries.

Reply:
I'm fine with users relying on `vcpkg` at runtime to satisfy their library dependencies. For things we build here, I'd rather we use Homebrew-provided (or system-provided) libraries and link with them dynamically.

`zlib` is provided by macOS, and never breaks their ABI. `libuv` follows semver, and should be safe to link with dynamically. (Recent security issues also make me hesitant to link statically here.) Homebrew `pcre2` carries some macOS-specific fixes that are typically not found in other builds. (This is actually true more generally about Homebrew-built libraries, but I know we can only go so far in using them.)